pkg/apis/etcd/v1beta2/cluster.go

```diff
@@ -64,6 +64,39 @@ func (c *EtcdCluster) AsOwner() metav1.OwnerReference {
 	}
 }
 
+type ProbeConfig struct {
+	InitialDelaySeconds int `json:"initialDelaySeconds,omitempty"`
+	TimeoutSeconds      int `json:"timeoutSeconds,omitempty"`
+	PeriodSeconds       int `json:"periodSeconds,omitempty"`
+	FailureThreshold    int `json:"failureThreshold,omitempty"`
+}
+
+type EtcdConfig struct {
```

Comment:
most of this parameters can be controlled directly via env vars, and etcd-opreator already supports in as `env` vars in .spec.pod.etcdEnv.

I would leave in here only k8s specific ones - liveness and readiness probes, but also would rather reconsider it and put that in pod spec.